extensions/cl_intel_device_attribute_query.asciidoc

```diff
@@ -0,0 +1,197 @@
+
+== New API Functions
+
+None.
+
+== New API Enums
+
+Possible values accepted as the _param_name_ parameter of *clGetDeviceInfo*, depending on the device type and the extension version.
+Additional queries may be added in subsequent versions of the extension:
+
+[source]
+----
+/* For GPU devices, version 1.0.0: */
+
+#define CL_DEVICE_IP_VERSION_INTEL                0x4250
+#define CL_DEVICE_ID_INTEL                        0x4251
+#define CL_DEVICE_NUM_SLICES_INTEL                0x4252
+#define CL_DEVICE_NUM_SUB_SLICES_PER_SLICE_INTEL  0x4253
+#define CL_DEVICE_NUM_EUS_PER_SUB_SLICE_INTEL     0x4254
+#define CL_DEVICE_NUM_THREADS_PER_EU_INTEL        0x4255
+#define CL_DEVICE_FEATURE_CAPABILITIES_INTEL      0x4256
+----
+
+Bitfield type describing the feature capabilities of a device.
+Additional feature flags may be added in subsequent versions of the extension.
+
+[source]
+----
+typedef cl_bitfield         cl_device_feature_capabilities_intel;
+
+/* For GPU devices, version 1.0.0: */
+
+#define CL_DEVICE_FEATURE_FLAG_DP4A_INTEL         (1 << 0)
+#define CL_DEVICE_FEATURE_FLAG_DPAS_INTEL         (1 << 1)
+----
+
+== New API Types
+
+None.
+
+== Modifications to the OpenCL API Specification
+
+(Add to the list preceding Table 5 in Section 4.2 - Querying Devices) ::
++
+--
+The device queries described in the Device Queries table should return the same information for a root-level device i.e. a device returned by *clGetDeviceIDs* and any sub-devices created from this device except for the following queries:
+
+* ...
+* {CL_DEVICE_NUM_SLICES_INTEL}
+--
+
+(Add to Table 5 - OpenCL Device Queries in Section 4.2 - Querying Devices) ::
++
+--
+[caption="Table 5. "]
+.List of supported param_names by clGetDeviceInfo
+[width="100%",cols="<30%,<20%,<50%",options="header"]
+|====
+| *cl_device_info* | Return Type | Description
+| {CL_DEVICE_IP_VERSION_INTEL}
+  | `cl_version`
+      | The IP version for the device.
+        The meaning of this version is implementation-defined, but newer devices should have a higher version than older devices.
+| {CL_DEVICE_ID_INTEL}
+  | `cl_uint`
+      | A unique device identifier for the OpenCL device.
+
+        If the implementation is driven primarily by a PCI device with a PCI device ID, the low 16 bits of the device identifier must contain that PCI device ID, and the remaining bits must be set to zero.
+        Otherwise, the choice of what to return may be dictated by operating system or platform policies - but should uniquely identify both the device version and any major configuration options (for example, core count in the case of multi-core devices).
+| {CL_DEVICE_NUM_SLICES_INTEL}
+  | `cl_uint`
+      | The number of slices in the device.
+        A slice is a collection of sub-slices.
+| {CL_DEVICE_NUM_SUB_SLICES_PER_SLICE_INTEL}
+  | `cl_uint`
+      | The maximum number of sub-slices in one slice.
+        A slice is a collection of execution units (EUs), fixed-function units, and caches.
```

Comment:
Slices are defined two different ways, while sub-slices aren't explicitly defined here. Maybe move this definition up to `CL_DEVICE_NUM_SLICE_INTEL` and here talk about what a sub-slice is? I'm not sure if it's relevant or not, but I was curious if the sub-slices are assumed or required to be the equivalent (ie, all contain the same numbers and types of units and caches).

Reply:
```suggestion
        A sub-slice is a collection of execution units (EUs), fixed-function units, and caches.
```
I had a similar thought as @jlewis-austin when reading this.  No idea if my suggestion is correct, but it seems to fit the context.

Reply:
Thank you for catching this, that's a little embarassing!

Alastair's suggested fix is correct.  I've applied it.